lib/main.js

```diff
@@ -38,23 +42,36 @@ export default {
       lintOnFly: true,
       lint: (activeEditor) => {
         const command = atom.config.get("linter-ruby.rubyExecutablePath");
+        const ignored = atom.config.get("linter-ruby.ignoredExtensions");
         const filePath = activeEditor.getPath();
-        return helpers.exec(command, ['-wc'], {stdin: activeEditor.getText(), stream: 'stderr'}).then(output => {
-          var toReturn = [];
-          output.split(/\r?\n/).forEach(function (line) {
-            const matches = regex.exec(line);
-            if (matches === null) {
-              return;
-            }
-            toReturn.push({
-              range: helpers.rangeFromLineNumber(activeEditor, Number.parseInt((matches[1] - 1))),
-              type: matches[2],
-              text: matches[3],
-              filePath: filePath
+        var ignore = false;
+
+        ignored.split(",").forEach(function (extension) {
+          if (filePath.split(".").pop() === extension.trim()) {
+            ignore = true;
+          };
+        });
+
+        if (ignore) {
```

Comment:
How about `if (ignore) return []`
You don't need an else block as you are returning from the first one.